authentication/http.go

```diff
@@ -42,6 +42,23 @@ func WithTenant(next http.Handler) http.Handler {
 	})
 }
 
+// WithTenantFromHeader finds the tenant from the HTTP headers and adds it to the request context.
+func WithTenantFromHeader(tenantHeader string) Middleware {
+	return func(next http.Handler) http.Handler {
+		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			tenant := r.Header.Get(tenantHeader)
+			if tenant == "" {
+				httperr.PrometheusAPIError(w, "tenant not found", http.StatusInternalServerError)
```

Comment:
I think we can make this `400` (if user does not provide the header, it's bad request)